crates/e2e/src/subxt_client.rs

```diff
@@ -527,6 +530,7 @@ where
         caller: &Keypair,
         message: &CallBuilderFinal<E, Args, RetType>,
         value: E::Balance,
+        extra_gas_portion: Option<usize>,
```

Comment:
> the old instantiate and call are left unchanged so this PR doesn't introduce breaking changes

This looks like a breaking change.

Not a huge problem for it to be a breaking change, just the fact that in 99% of cases the value passed will be `None`.

One approach would be to have a new method e.g. `bare_call` which just passes raw arguments to (in this case) `self.api.call()`. The `gas_limit` would be an absolute value and the part of adding the "portion" could just be done in the builder exec method.

Then we could even keep this method as is so it is a non-breaking change if we wanted. But this is not strictly necessary since we are doing a major release...it will be just annoying for users to have to update all their tests.

Reply:
>  The gas_limit would be an absolute value and the part of adding the "portion" could just be done in the builder exec method.

I am not fun of an idea of having 3 different ways of building a call. We are going to end up with `call()`, `bare_call()` and call builder which serve similar purposes. This will just confuse people

Reply:
Additionally, there is no way of specifying gas limit in E2E anyway, so it will be a breaking change nonetheless